modules/imaging_uploader/php/imaging_uploader.class.inc

```diff
@@ -662,13 +677,47 @@ class Imaging_Uploader extends \NDB_Menu_Filter_Form
      */
     function toJSON()
     {
+        $pscToCandID = array();
+        $candToPSCID = array();
+
+        foreach ($this->IDs as $ID) {
+            $p = $ID['PSCID'];
+            $c = $ID['CandID'];
+
+            $pscToCandID[$p] = $c;
+            $candToPSCID[$c] = $p;
+        }
+
         $result            = $this->toArray();
         $result['form']    = $this->form->form;
         $result['mriList'] = $this->_getFileList($result['Data']);
+        $result['candToPSCID'] = $candToPSCID;
+        $result['pscToCandID'] = $pscToCandID;
 
         $result['maxUploadSize'] = \Utility::getMaxUploadSize();
 
         return json_encode($result);
     }
+
+    /**
+     * Returns candidate PSCID / CandID array
+     *
+     * @return array $IDs database return of candidate IDs
+     */
+    function getCandidateIDs()
+    {
+        $db   = \Database::singleton();
+        $user = \User::singleton();
+        $IDs  = $db->pselect(
+            "SELECT PSCID, CandID ".
+            "FROM candidate ".
+            "WHERE FIND_IN_SET(CenterID, :cids) ".
+            "AND Entity_type='Human' " .
+            "ORDER BY PSCID",
+            array('cids' => implode(',', $user->getCenterIDs()))
```

Comment:
if the user uploading is affiliated with the scanning site only, he/she won't be able to see the candidates at the recruitment site. The current behavior of Imaging Uploader does not have the "own site" restriction; which I assume this function indirectly implements.